bin/initOSProjects.sh

```diff
@@ -35,3 +35,21 @@ for project in ${PROJECT_NAMESPACE}-${DEV} ${PROJECT_NAMESPACE}-${TEST} ${PROJEC
   exitOnError
 
 done
+
+#look through the tools env for artifactory-creds
+#setup artifactory/docker pull creds
+USE_PULL_CREDS=${USE_PULL_CREDS:-true}
+CRED_SEARCH_NAME=${CRED_SEARCH_NAME:-artifacts-default}
+PULL_CREDS=${PULL_CREDS:-artifactory-creds}
+DOCKER_REG=${DOCKER_REG:-docker-remote.artifacts.developer.gov.bc.ca}
+CRED_ENVS="${CRED_ENVS[@]:-tools dev test prod}"
+PROMPT_CREDS=${PROMPT_CREDS:-false}
+
+
+if [ ! -z ${USE_PULL_CREDS} ] && [ ${USE_PULL_CREDS} = true ]; then
```

Comment:
Combine buildPromptCreds and buildCreds in such a way that the user does not have to decide which call to make.  The user should only have to make a single call with the information available.  The functions should figure out what to do from there as they know what information they need for the various options.

Also rename the functions to better describe their purpose; `registerPullSecret`.

Reply:
@WadeBarnes so the way it works right now is that if CRED_SEARCH_NAME is not found then the script simply ends. If the user specifies PROMPT_CREDS, only then will the user be prompted for docker credentials. Should I intead make it so that whenever CRED_SEARCH_NAME is not found I promt the user for creds?
Also, updated the function names

Reply:
I'm saying push this logic down into a function that combines `buildPromptCreds` and `buildCreds` (now `registerPullSecretPrompt` and `registerPullSecret`) so the user only needs to make a single function call at this level.  It makes the code cleaner and more maintainable.

Reply:
change completed, pushed registerPullSecretPrompt and registerPullSecret to one function that determines what to do